src/txn/txn_timestamp.c

```diff
@@ -86,8 +86,8 @@ __wt_verbose_timestamp(WT_SESSION_IMPL *session,
  *	Decodes and sets a timestamp.
  */
 int
-__wt_txn_parse_timestamp(WT_SESSION_IMPL *session,
-     const char *name, wt_timestamp_t *timestamp, WT_CONFIG_ITEM *cval)
+__wt_txn_parse_timestamp(WT_SESSION_IMPL *session, const char *name,
+    wt_timestamp_t *timestamp, WT_CONFIG_ITEM *cval, bool zero_ok)
```

Comment:
I don't love having to change all callers of this function for one case, could you split it up into multiple functions instead? Maybe a new `__wt_txn_parse_timestamp_zero_ok` that is called by `__wt_txn_parse_timestamp`, or a shared internal function?